Tracking client: update MLFlow run status when we update the status tag

peteski22 opened the pull request:
# What's changing

* Updates the status of the run in MLFlow as well as the tag we create for status.

# How to test it

Steps to test the changes:

1.
2.
3.

# Additional notes for reviewers

This is the 'status' shown in MLFlow when we don't come back to update it at the end:

![image](https://github.com/user-attachments/assets/9a743e23-5fec-43ec-9802-8eeaca208a03)

Now the status no longer says 'Running'

![image](https://github.com/user-attachments/assets/0628dd9f-c522-411f-9450-8db75caa9794)

So we can see what was successful/failed in the MLFlow dashboard too:

![image](https://github.com/user-attachments/assets/954398a2-72a7-4c13-9225-468006541ad3)

# I already...

- [ ] Tested the changes in a working environment to ensure they work as expected
- [ ] Added some tests for any new functionality
- [ ] Updated the documentation (both comments in code and [product documentation](https://mozilla-ai.github.io/lumigator) under `/docs`)
- [ ] Checked if a (backend) DB migration step was required and included it if required

njbrake reviewed (approved):
LGTM! I'm not sure whether we set update_workflow_status to take advantage of those workflowstatus' that you add, but I trust that you're aware of the plan and this is a step in the process. Thanks :)

peteski22 commented:
> LGTM! I'm not sure whether we set update_workflow_status to take advantage of those workflowstatus' that you add, but I trust that you're aware of the plan and this is a step in the process. Thanks :)

😄 I added some screenshot receipts to help explain that it's fixing (the Status not being updated in MLFlow), although we were tracking the 'status' as a tag we added manually, it now lines up.
